uaclient/util.py

```diff
@@ -497,6 +497,54 @@ def configure_web_proxy(
         request.install_opener(opener)
 
 
+def configure_snap_proxy(
+    http_proxy: "Optional[str]", https_proxy: "Optional[str]"
+) -> None:
+    """
+    Configure snap to use http and https proxies.
+
+    :param http_proxy: http proxy to be used by snap. If None, it will
+                       not be configured
+    :param https_proxy: https proxy to be used by snap. If None, it will
+                        not be configured
+    """
+    if http_proxy:
+        subp(["snap", "set", "system", "proxy.http={}".format(http_proxy)])
+
+    if https_proxy:
+        subp(["snap", "set", "system", "proxy.https={}".format(https_proxy)])
+
+
+def configure_livepatch_proxy(
+    http_proxy: "Optional[str]", https_proxy: "Optional[str]"
+) -> None:
+    """
+    Configure livepatch to use http and https proxies.
+
+    :param http_proxy: http proxy to be used by livepatch. If None, it will
+                       not be configured
+    :param https_proxy: https proxy to be used by livepatch. If None, it will
+                        not be configured
+    """
+    if http_proxy:
+        subp(
+            [
+                "canonical-livepatch",
+                "config",
+                "http-proxy={}".format(http_proxy),
```

Comment:
I'm not as worried about retries here, because we call this only after we've already retried and waited to install livepatch. And we haven't yet needed retries on uaclient.entitlements.livepatch.process_config_directives in the past so we are unlikely to hit a race calling other livepatch config commands.